### PR TITLE
raidboss: add support for P12S Engravements

### PR DESCRIPTION
**Pull request by wexxlee**
Not sure if I did something dumb while rebasing or Visual Studio did, but it was a messy unwind, and doing a new PR was POLR.  So I'm killing the old PR (#5580 ) and opening a new one.  This includes all revisions incorporating feedback from the initial code review.  Apologies for any added work.

### CODE REVIEW

File: ui/raidboss/data/06-ew/raid/p12s.ts [hunk 1]
```diff
@@ -114,6 +152,14 @@ export interface Data extends RaidbossData {
...
   isDoorBoss: boolean;
   phase?: 'superchain1' | 'palladion' | 'superchain2a' | 'superchain2b';
+  engravementCounter: number;
+  engravement1Towers: string[];
+  engravement2MyLabel?: EngravementLabel;
+  engravement3TowerType?: 'lightTower' | 'darkTower';
+  engravement3TowerPlayers: string[];
+  engravement3TetherPlayers: { [name: string]: AnthroposTether };
+  engravement3DarkTethersSide?: 'east' | 'west';
+  engravement3LightTethersSide?: 'east' | 'west';
```
Comment (quisquous) [inline ui/raidboss/data/06-ew/raid/p12s.ts:162]: Are these redundant? Do we only need one of these variables?

File: ui/raidboss/data/06-ew/raid/p12s.ts [hunk 2]
```diff
@@ -508,6 +558,386 @@ const triggerSet: TriggerSet<Data> = {
...
+      outputStrings: {
+        lightTilt: {
+          en: 'Soak dark tower',
+        },
+        darkTilt: {
+          en: 'Soak light tower',
+        },
+      },
+    },
+    // In Engravement 2 (Superchain 1), all supports or  DPS will receive lightTilt and darkTilt (2 each).
```
Comment (quisquous) [inline ui/raidboss/data/06-ew/raid/p12s.ts:616]:
```suggestion
    // In Engravement 2 (Superchain 1), all supports or DPS will receive lightTilt and darkTilt (2 each).
```

File: ui/raidboss/data/06-ew/raid/p12s.ts [hunk 3]
```diff
@@ -508,6 +558,386 @@ const triggerSet: TriggerSet<Data> = {
...
+    // The other role group will receive lightTower, darkTower, lightBeam, and darkBeam.
+    // To resolve the Beams during the 2nd orb, lightBeam needs to stack with darkTower and both darkTilts, and vice versa.
+    // After the 3rd orb, lightTower and darkTower will drop their towers, and  darkBeam and lightBeam (respectively) will soak them.
+    // The four Heavensflame players all simultaneously need to spread to drop their AoEs.
+    // Debuffs do change based on mechanic resolution, which can complicate things:
+    // - When a lightTilt player soaks a dark beam, their debuff will change to darkTilt, and vice versa.
+    // - Once the beams detonate, the lightBeam debuff disappears and is replaced with lightTilt (same with dark).
+    // So only use the initial debuff to resolve the mechanic, and use a long suppress to avoid incorrect later alerts.
+    {
+      id: 'P12S Engravement 2 Debuff', // Add Spread Reminder
```
Comment (quisquous) [inline ui/raidboss/data/06-ew/raid/p12s.ts:627]:
```suggestion
      id: 'P12S Engravement 2 Debuff',
```

^ was this a TODO?

**Review by quisquous — COMMENTED** [review]
Looks good, thanks! I will give this a spin tomorrow and see how it feels.

File: ui/raidboss/data/06-ew/raid/p12s.ts (hunk 3 above)
Comment (wexxlee) [reply ui/raidboss/data/06-ew/raid/p12s.ts:627]: internal note i forgot to remove 😄

File: ui/raidboss/data/06-ew/raid/p12s.ts (hunk 1 above)
Comment (wexxlee) [reply ui/raidboss/data/06-ew/raid/p12s.ts:162]: It looks like lightTower/darkTower are applied ~1.1s before the adds begin casting, so I think we could just collect the TetherSide that corresponds to the same color as the tower debuff, and use that to resolve the Towers Drop Location.

Comment (quisquous) [reply ui/raidboss/data/06-ew/raid/p12s.ts:162]: Mostly I was just saying that the dark tethers are one side and the light tethers therefore must be on the opposite side.  So, you only need to track one of them to know where they both are.

Comment (wexxlee) [reply ui/raidboss/data/06-ew/raid/p12s.ts]: That is...also fair.  I ended up doing it this way and cut some code in the process, so 🤷‍♂️

**Comment by wexxlee** [comment]
> Looks good, thanks! I will give this a spin tomorrow and see how it feels.

Cool.  I have a handful of other small p12s p1 triggers ready to merge as well (Para1 clones N/S, safe lanes for add cleaves during Para2, Parthenos line cleave during SC IIB, and the Glaukopis tank swap).  Would you prefer to have them added to this PR, or wait to land this first?

**Comment by YuzukiTsuru** [comment]
Great work, I also add some p12s p1 triggers like P12S Engravement 1 Tower Drop notify which direction will go, and Beam type triggers base on this PR. Waiting for the merge.

**Comment by quisquous** [comment]
Thanks for all the changes! This seemed great in practice.

**Pull request merged**